File: src/main/java/org/goafabric/catalog/controller/ConditionController.java
package org.goafabric.catalog.controller;

import org.goafabric.catalog.job.condition.ConditionEo;
import org.goafabric.catalog.job.condition.ConditionRepository;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "conditions", produces = MediaType.APPLICATION_JSON_VALUE)
public class ConditionController {
    private final ConditionRepository repository;

    public ConditionController(ConditionRepository repository) {
        this.repository = repository;
    }

    @GetMapping("/{id}")
    public ConditionEo getById(@PathVariable String id) {
        return repository.findById(id).orElseThrow();
    }


    @GetMapping("/findByCode")
    public ConditionEo findByCode(@RequestParam("code") String code) {
        return repository.findByCode(code);
    }

    @GetMapping("/findByDisplay")
    public List<ConditionEo> findByDisplay(@RequestParam("display") String display) {
        return repository.findByDisplayStartsWith(display);
    }
}

File: src/main/kotlin/org/goafabric/catalog/controller/ConditionController.kt
package org.goafabric.catalog.controller

import org.goafabric.catalog.job.condition.ConditionEo
import org.goafabric.catalog.job.condition.ConditionRepository
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["conditions"], produces = [MediaType.APPLICATION_JSON_VALUE])
class ConditionController(private val repository: ConditionRepository) {
    @GetMapping("/{id}")
    fun getById(@PathVariable id: String): ConditionEo {
        return repository.findById(id).orElseThrow()!!
    }


    @GetMapping("/findByCode")
    fun findByCode(@RequestParam("code") code: String): ConditionEo {
        return repository.findByCode(code)
    }

    @GetMapping("/findByDisplay")
    fun findByDisplay(@RequestParam("display") display: String): MutableList<ConditionEo> {
        return repository.findByDisplayStartsWith(display)
    }
}
